### PR TITLE
docker.com/cli

### LINKED ISSUE #277: +docker (333/548)
Formulae: https://formulae.brew.sh/formula/docker

### PR DESCRIPTION
**Pull request by and-ri**
closes #277

### CODE REVIEW

**Comment by and-ri** [comment]
Hmm, I don’t understand what’s wrong with colima ... I checked it on several macbooks and everything starts on them 🧐

**Comment by jhheider** [comment]
Any chance it's conflicting with the running docker instance on the builders?

**Comment by and-ri** [comment]
> Any chance it's conflicting with the running docker instance on the builders?

Oh, I didn’t even think about the fact that the docker might already be running on the builder 🫣

**Comment by and-ri** [comment]
I'm not sure if we can just leave the version check in that case?
And I do not fully understand what is meant by "stub the binaries" here https://github.com/teaxyz/pantry/issues/277#issuecomment-1643900009

**Comment by jhheider** [comment]
In this instance, I suspect he means write a shim that will call docker desktop binaries, if present, otherwise use tea-provided colima in their place (not yet packaged).

**Comment by mxcl** [comment]
instead of stubbing we should probs just document it. I've started a thing where I add a `README.md` to the gui which we can put things like this.

File: projects/docker.com/cli/package.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+distributable:
+  url: https://github.com/docker/cli/archive/v{{version}}.tar.gz
+  strip-components: 1
+versions:
+  github: docker/cli/tags
+platforms:
+  # colima -> only darwin
+  - darwin
```
Comment (mxcl) [inline projects/docker.com/cli/package.yml]: we should build for linux and make the dependency darwin only?

and well, the dep isn't a dep really. It's a `companion` at best.

companions cannot be platform restricted tho.

will do for now to have it as a dep. but yeah we should build for linux if poss.

**Comment by and-ri** [comment]
> instead of stubbing we should probs just document it. I've started a thing where I add a `README.md` to the gui which we can put things like this.

The ability to add a `README.md` would be great
The README file will be placed right next to the package.yml, am I understanding correctly?

**Comment by mxcl** [comment]
> The README file will be placed right next to the package.yml, am I understanding correctly?

yes, if you take a look you can see a few I've added

File: projects/docker.com/cli/package.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+distributable:
+  url: https://github.com/docker/cli/archive/v{{version}}.tar.gz
+  strip-components: 1
+versions:
+  github: docker/cli/tags
+dependencies:
+  darwin:
+    github.com/abiosoft/colima: '*'
```
Comment (mxcl) [inline projects/docker.com/cli/package.yml]: this should be removed pls

there is no mandatory dep

File: projects/docker.com/cli/package.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          go-md2man -in="$md" -out="{{prefix}}/man/man${section}/$(basename "$md" .md)"
+        done
+  env:
+    GOPATH: $PWD
+    GO111MODULE: auto
+    ARGS:
+      - -trimpath
+      - -o={{prefix}}/bin/docker
+provides:
+  - bin/docker
```
Comment (mxcl) [inline projects/docker.com/cli/package.yml:32]: doesn't provide more?

**Review by mxcl — CHANGES_REQUESTED** [review]

File: projects/docker.com/cli/package.yml (hunk 2 above)
Comment (and-ri) [reply projects/docker.com/cli/package.yml]: 👌🏻

File: projects/docker.com/cli/package.yml (hunk 3 above)
Comment (and-ri) [reply projects/docker.com/cli/package.yml:32]: only `docker`
image

File: projects/docker.com/cli/package.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+test:
+  script:
+    - docker --version | grep {{version}}
```
Comment (mxcl) [inline projects/docker.com/cli/package.yml:35]: ideally we'd start colima and test it properly, but this is fine for now

**Review by mxcl — APPROVED** [review]

**Pull request merged**